### PR TITLE
friendship page now has the option to show settled transactions

### PR DESCRIPTION
**Pull request by Bbowen100**

### CODE REVIEW

File: app/components/friendship-page/component.js [hunk 1]
```diff
@@ -21,6 +23,21 @@ export default Ember.Component.extend({
     return this.get('userTransactions').sortBy('createdAt').reverse()
   }.property('userTransactions.[]'),
 
+  userSettledTransactions: function() {
+    let relatedObjectId = this.get('friendship.friendshipDataId')
+
+    let settledTransactions = this.get('store').filter('transaction', t => {
+      console.log(t.get('memo'))
```
Comment (ch-maazali) [inline app/components/friendship-page/component.js]: Should remove console.log

Comment (Bbowen100) [inline app/components/friendship-page/component.js]: Oh shit ...

File: app/models/transaction.js [hunk 2]
```diff
@@ -11,7 +11,8 @@ let Transaction = DS.Model.extend({
   creator: DS.belongsTo('user'),
   createdAt: DS.attr('Date', {
     defaultValue: function() { return new Date() }
-  })
+  }),
+  is_settled: DS.attr('boolean')
```
Comment (MaazAli) [inline app/models/transaction.js]: property name should be camelCase. (`isSettled`)

File: app/components/friendship-page/component.js [hunk 3]
```diff
@@ -21,6 +23,20 @@ export default Ember.Component.extend({
     return this.get('userTransactions').sortBy('createdAt').reverse()
   }.property('userTransactions.[]'),
 
+  userSettledTransactions: function() {
+    let relatedObjectId = this.get('friendship.friendshipDataId')
+
+    let settledTransactions = this.get('store').filter('transaction', t => {
+      return t.get('relatedObjectId') == relatedObjectId &&
+      t.get('relatedObjectType') == 'Friendship' && t.get('is_settled') == true
```
Comment (MaazAli) [inline app/components/friendship-page/component.js]: use `t.get('isSettled')` property name should be `camelCase`.

File: app/components/friendship-page/component.js [hunk 4]
```diff
@@ -21,6 +23,20 @@ export default Ember.Component.extend({
...
+    })
+    return settledTransactions
+  }.property('friendship.friendshipDataId'),
+
+  sortedUserSettledTransactions: function() {
+    return this.get('userSettledTransactions').sortBy('createdAt').reverse()
+  }.property('userTransactions.[]'),
```
Comment (MaazAli) [inline app/components/friendship-page/component.js]: `sortedUserSettledTransactions` should be computed off of `userSettledTransactions` and not `userTransactions`.

**Comment by MaazAli** [comment]
Tested it out and it works 👍 . I did leave some comments that are mostly stylistic. 

What was the need to create a separate component for the `settled-list`. Why couldn't we just expand the `transaction-list` component to accomodate for this?

Also, the current main transaction list shows all transactions, we'd like to filter that to only show un settled transactions. If users want to see settled transactions, they can open up the collapsed list.

**Comment by Bbowen100** [comment]
What was the need to create a separate component for the settled-list. Why couldn't we just expand the transaction-list component to accomodate for this?
  - That is what I did originally but thought it would be more modular and easier to understand if I made a separate component.

Also, the current main transaction list shows all transactions, we'd like to filter that to only show un settled transactions. If users want to see settled transactions, they can open up the collapsed list.
 - On it.

File: tests/integration/components/settled-list-test.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { moduleForComponent, test } from 'ember-qunit';
```
Comment (MaazAli) [inline tests/integration/components/settled-list-test.js]: We don't need this file anymore.

File: app/components/friendship-page/template.hbs [hunk 6]
```diff
@@ -42,4 +42,17 @@
     transactions=sortedUserTransactions
     transactionUpdated="updateBalance"
   }}
+
+  <button class="btn settle-transaction" {{action 'wantsSettled'}} style="width:100%">
```
Comment (MaazAli) [inline app/components/friendship-page/template.hbs]: To support other full-width buttons. I recommend adding a class under this: https://github.com/tbbr/tbbr-client/blob/master/app/styles/base.scss#L127

Something like this 

```
.btn---full-width {
    width: 100%
}
```

and then you can just have 
`<button class="btn btn--full-width" {{action 'wantsSettled'}}>`

Comment (MaazAli) [inline app/components/friendship-page/template.hbs]: Are we still using the `settle-transaction` css class?

**Comment by MaazAli** [comment]
@Bbowen100 minor changes recommended. Good work on the PR.

File: app/components/friendship-page/styles.scss [hunk 7]
```diff
@@ -53,14 +53,10 @@ $addTransactionBtnRightPos: 20px;
       padding: 0 15px;
       width: 64px
     }
+  }
 
-    .btn.settle-transaction {
```
Comment (MaazAli) [inline app/components/friendship-page/styles.scss]: We're using this class else where, don't need to remove it.

**Comment by MaazAli** [comment]
Looks good 👍

**Pull request merged**